protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -253,16 +271,30 @@ func (fc *FinalizationConsensus) ExpectedBlockHeight(chainParser chainlib.ChainP
 		}
 		return lavaslices.Median(data)
 	}
+
 	medianOfExpectedBlocks := median(mapExpectedBlockHeights)
 	providersMedianOfLatestBlock := medianOfExpectedBlocks + int64(blockDistanceForFinalizedData)
 	if debug {
-		utils.LavaFormatDebug("finalization information", utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "mapExpectedBlockHeights", Value: mapExpectedBlockHeights}, utils.Attribute{Key: "medianOfExpectedBlocks", Value: medianOfExpectedBlocks}, utils.Attribute{Key: "latestBlock", Value: fc.latestBlockByMedian}, utils.Attribute{Key: "providersMedianOfLatestBlock", Value: providersMedianOfLatestBlock})
+		utils.LavaFormatDebug("finalization information",
+			utils.LogAttr("specId", fc.specId),
+			utils.LogAttr("mapExpectedBlockHeights", mapExpectedBlockHeights),
+			utils.LogAttr("medianOfExpectedBlocks", medianOfExpectedBlocks),
+			utils.LogAttr("latestBlock", fc.prevLatestBlockByMedian),
+			utils.LogAttr("providersMedianOfLatestBlock", providersMedianOfLatestBlock),
+		)
 	}
-	if medianOfExpectedBlocks > 0 && uint64(providersMedianOfLatestBlock) > fc.latestBlockByMedian {
-		if uint64(providersMedianOfLatestBlock) > fc.latestBlockByMedian+1000 && fc.latestBlockByMedian > 0 {
-			utils.LavaFormatError("uncontinuous jump in finalization data", nil, utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "s.prevEpochProviderHashesConsensus", Value: fc.prevEpochProviderHashesConsensus}, utils.Attribute{Key: "s.currentProviderHashesConsensus", Value: fc.currentProviderHashesConsensus}, utils.Attribute{Key: "latestBlock", Value: fc.latestBlockByMedian}, utils.Attribute{Key: "providersMedianOfLatestBlock", Value: providersMedianOfLatestBlock})
+
+	if medianOfExpectedBlocks > 0 && uint64(providersMedianOfLatestBlock) > fc.prevLatestBlockByMedian {
+		if uint64(providersMedianOfLatestBlock) > fc.prevLatestBlockByMedian+1000 && fc.prevLatestBlockByMedian > 0 {
+			utils.LavaFormatError("uncontinuous jump in finalization data", nil,
+				utils.LogAttr("specId", fc.specId),
+				utils.LogAttr("prevEpochBlockToHashesToAgreeingProviders", fc.prevEpochBlockToHashesToAgreeingProviders),
+				utils.LogAttr("currentBlockToHashesToAgreeingProviders", fc.currentBlockToHashesToAgreeingProviders),
+				utils.LogAttr("latestBlock", fc.prevLatestBlockByMedian),
+				utils.LogAttr("providersMedianOfLatestBlock", providersMedianOfLatestBlock),
+			)
 		}
-		atomic.StoreUint64(&fc.latestBlockByMedian, uint64(providersMedianOfLatestBlock)) // we can only set conflict to "reported".
+		fc.prevLatestBlockByMedian = uint64(providersMedianOfLatestBlock) // we can only set conflict to "reported".
```

Comment:
cannot write under a read lock! 🔢

Reply:
return the atomics! and do a swift check you didn't make any of those mistakes else where.

Reply:
Did it! Also, this is the only write to this field